viewer/packages/camera-manager/src/DefaultCameraManager.ts

```diff
@@ -554,7 +556,13 @@ export class DefaultCameraManager implements CameraManager {
 
       if (wantNewScrollTarget && isZoomToCursor) {
         scrollStarted = true;
+
+        // Disable controls to prevent camera from moving while picking is happening.
+        // await is not working as expected because event itself is not awaited.
+        this._controls.temporaryDisable(false);
         const newTarget = await this.calculateNewTarget(e);
+        this._controls.temporaryDisable(this._controls.enabledCopy);
```

Comment:
This doesn't work in all cases, unfortunately. It was exactly the same before without any modifications for ComboControls as well. All this mess is required because two wheel events are issued at the same time and when one event disabled controls, another one reads it and then returns the state of controls to disabled.

Reply:
Ok, I see. I think a better fix is to temporarily stop listening to new events, rather than keeping state. This should be done by unregistering from the event listener whenever onWheel fires, and listening again when it finishes. Do it in a `try { ... } finally { ... }` block

Reply:
It can probably be done in the same place as here actually.

Reply:
It also is not working as expected because it causes "lag" for user interaction. Tried doing that with just a flag for cancelling events that are coming at the same time. Pushed new version now so that you can take a look.